### PR TITLE
Build all cookbook and benchmark plugins when validating input files.

### PR DESCRIPTION
**Pull request by gassmoeller**
Simplified version of #5319 that only builds all plugins. This would still fix #5259 and make it possible to remove some special rules.

### CODE REVIEW

File: benchmarks/check.mk [hunk 1]
```diff
@@ -74,15 +76,6 @@ main: dummy
 # example/: dummy
 #	@$(def); run_prm $@ test.prm
 
-annulus/: dummy
-	+@$(def); make_lib $@/plugin
-	@$(def); run_all_prms $@/instantaneous
```
Comment (tjhei) [inline benchmarks/check.mk:79]: you need to keep this rule because by default only */*.prm are run, right?

File: benchmarks/check.mk [hunk 2]
```diff
@@ -124,22 +113,10 @@ newton_solver_benchmark_set/: dummy tosi_et_al_2015_gcubed/
 onset-of-convection/: dummy
 	@echo "TODO"
 
-operator_splitting/: dummy
-	+@$(def); make_lib $@/advection_reaction
```
Comment (tjhei) [inline benchmarks/check.mk:128]: same here

File: benchmarks/check.mk [hunk 3]
```diff
@@ -124,22 +113,10 @@ newton_solver_benchmark_set/: dummy tosi_et_al_2015_gcubed/
...
-	@$(def); run_all_prms $@/exponential_decay
-
 rayleigh_taylor_instability/: dummy
 	+@$(def); make_lib $@
 	@$(def); run_prm $@ rayleigh_taylor_instability.prm
 
-rigid_shear/: dummy
-	+@$(def); make_lib $@/plugin
-	@$(def); run_all_prms $@/instantaneous
-	@$(def); run_all_prms $@/steady-state
```
Comment (tjhei) [inline benchmarks/check.mk:140]: same here

**Review by tjhei — APPROVED** [review]

**Comment by gassmoeller** [comment]
Ok, updated. But please wait with merging until #5259 is merged to not create conflicts.

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
Ready to merge whenever you would like.

**Pull request merged**